rattler_pty fails to build on illumos

### Checklist

- [x] I added a descriptive title
- [x] I searched open reports and couldn't find a duplicate

### What happened?

The issue is quite similar to the https://github.com/conda/rattler/issues/2337 - I was building the same `mise` app on the SmartOS distribution of the illumos.

Observed error:
```
...
   Compiling rattler_pty v0.2.15
error[E0425]: cannot find function `ptsname_r` in this scope
   --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rattler_pty-0.2.15/src/unix/pty_process.rs:148:26
    |
148 |         let slave_name = ptsname_r(&master_fd)?;
    |                          ^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
```

illumos code branch is non-existant in the `nix` crate
https://github.com/nix-rust/nix/blob/v0.31.0/src/pty.rs#L217

so the shortest path is to extend `NetBSD` branch in the `rattler`
https://github.com/conda/rattler/blob/main/crates/rattler_pty/src/unix/pty_process.rs#L30
as the `libc` crate exposes `ptsname_r` for the illumos too.

### Additional Context

```
> uname -a
SunOS saule 5.11 joyent_20260710T055317Z i86pc i386 i86pc illumos
```
----
```
> man -k ptsname
ptsname(3c), ptsname_r(3c) - get the name of the subsidiary device of a pseudo-terminal
```
https://illumos.org/man/3C/ptsname / https://illumos.org/man/3C/ptsname_r
compare with the NetBSD
https://man.netbsd.org/ptsname.3

----
`libc` crate:
https://docs.rs/libc/latest/x86_64-unknown-illumos/libc/fn.ptsname.html
https://docs.rs/libc/latest/x86_64-unknown-illumos/libc/fn.ptsname_r.html

--- a/crates/rattler_pty/src/unix/pty_process.rs
+++ b/crates/rattler_pty/src/unix/pty_process.rs
@@ -27,9 +27,10 @@ use tokio::{
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use nix::pty::ptsname_r;
 
-#[cfg(target_os = "netbsd")]
+#[cfg(any(target_os = "netbsd", target_os = "illumos"))]
 /// NetBSD has `ptsname_r` in libc but apparently the `nix` crate does not expose it for NetBSD.
 /// Call `libc::ptsname_r` directly.
+/// The same is applicable for illumos
 fn ptsname_r(fd: &PtyMaster) -> nix::Result<String> {
     use std::ffi::CStr;
 

